### PR TITLE
Add more 'readonly' tags to the schema for fields not intended to be editable by users

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
This continues work to make our schema more useful to metadata-based forms (Afform & SearchKit).

Notes
----------------
The 'readonly' tag is new in our schema, so adding them doesn't affect any existing core code or extensions.

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
api_v3_SyntaxConformanceTest::testCreateSingleValueAlter with data set #82 ('UFField')
Failure in api call for UFField create:  Undefined index: type

**Comment by eileenmcnaughton** [comment]
test this please

File: xml/schema/Contact/Contact.xml [hunk 1]
```diff
@@ -594,6 +595,7 @@
     <type>varchar</type>
     <length>255</length>
     <comment>Cache Postal greeting.</comment>
+    <readonly>true</readonly>
```
Comment (eileenmcnaughton) [inline xml/schema/Contact/Contact.xml:598]: These fields are actually UI selectable - ie when you go to edit them from contact summary you can choose a different option value

image

Comment (eileenmcnaughton) [reply xml/schema/Contact/Contact.xml:598]: (Having trouble screenshotting the above in edit mode)

Comment (colemanw) [reply xml/schema/Contact/Contact.xml:598]: Are yo sure you aren't thinking of the `postal_greeting_id` field?

Comment (eileenmcnaughton) [reply xml/schema/Contact/Contact.xml:598]: Looks like it - they are so clear those field too....

**Pull request merged**